### PR TITLE
Added indent and newLines options

### PR DESCRIPTION
**Pull request by bad-ash**

### CODE REVIEW

File: templates.ts [hunk 1]
```diff
@@ -64,6 +64,9 @@ export function generateRendererFile(viewHash: string, viewFunction: string, dec
     } else {
         initBody = initBodyWithoutDecoder(viewHash, viewFunction);
     }
+    let newLinesStr = newLines !== undefined ? (newLines.toString().charAt(0).toUpperCase() + newLines.toString().slice(1)) : "True";
```
Comment (eeue56) [inline templates.ts]: this is really hard to read, and i'm not sure why it's so complicated?

Comment (bad-ash) [reply templates.ts]: it's just checking to see if newLines was specified, and if it was then capitalizes the first letter cause bools in elm start with an uppercase letter

Comment (eeue56) [reply templates.ts]: I would just make it like this: `newLineStr = newLines ? "True" : "False"`. Much simpler :)

File: templates.ts [hunk 2]
```diff
@@ -64,6 +64,9 @@ export function generateRendererFile(viewHash: string, viewFunction: string, dec
     } else {
         initBody = initBodyWithoutDecoder(viewHash, viewFunction);
     }
+    let newLinesStr = newLines !== undefined ? (newLines.toString().charAt(0).toUpperCase() + newLines.toString().slice(1)) : "True";
+    let indentStr = indent !== undefined ? indent.toString() : "4";
```
Comment (eeue56) [inline templates.ts]: no need for toString here

File: templates.ts [hunk 3]
```diff
@@ -64,6 +64,9 @@ export function generateRendererFile(viewHash: string, viewFunction: string, dec
     } else {
         initBody = initBodyWithoutDecoder(viewHash, viewFunction);
     }
+    let newLinesStr = newLines !== undefined ? (newLines.toString().charAt(0).toUpperCase() + newLines.toString().slice(1)) : "True";
+    let indentStr = indent !== undefined ? indent.toString() : "4";
+    let optionsSet = "options = { defaultFormatOptions | newLines = " + newLinesStr +", indent = " + indentStr + " }";
```
Comment (eeue56) [inline templates.ts]: Use 
```
`options { newLines = ${newLines}, indent = ${indent}`
```

instead

Comment (bad-ash) [reply templates.ts]: you mean that is what optionsSet should be assigned? what about the defaultFormatOptions?

File: templates.ts (hunk 1 above)
Comment (bad-ash) [reply templates.ts]: it doesn't work unless newLines/indent is explicitly checked against undefined

File: templates.ts (hunk 2 above)
Comment (bad-ash) [reply templates.ts]: hmm but typescript throws an error when i remove toString?

File: templates.ts (hunk 3 above)
Comment (eeue56) [reply templates.ts]: I mean use template literals: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: templates.ts (hunk 1 above)
Comment (eeue56) [reply templates.ts]: Right, but I mean make it just "True" or "False". Don't do this complicated approach of modifying the string -- go for the simplest approach.

File: templates.ts [hunk 4]
```diff
@@ -65,6 +65,18 @@ export function generateRendererFile(viewHash: string, viewFunction: string, dec
...
 
+    let newLinesStr;
+    if(newLines === undefined || newLines === true) {
+      newLinesStr = "True";
+    }
+    else {
+      newLinesStr = "False";
+    }
+
+    let indentStr = indent !== undefined ? indent : "4";
```
Comment (eeue56) [inline templates.ts]: this should just be 4 rather than the string 4

**Comment by eeue56** [comment]
Published as `0.0.9-alpha.2`. Thanks!

**Comment by bad-ash** [comment]
sweet, no problem

**Pull request merged**